libvips/foreign/magick.c

```diff
@@ -102,6 +105,54 @@ magick_sniff( const unsigned char *bytes, size_t length )
 		 bytes[2] == 11) )
 		return( "TGA" );
 
+	guint32 ifd_offset = 0;
+
+	/* Big-endian TIFF header
+	 */
+	if( length >= 8 &&
+		bytes[0] == 'M' &&
+		bytes[1] == 'M' &&
+		bytes[2] == 0 &&
+		bytes[3] == 42 )
+		ifd_offset = ((guint32) bytes[4] << 24) |
+			((guint32) bytes[5] << 16) |
+			((guint32) bytes[6] << 8) |
+			bytes[7];
+
+	/* Little-endian TIFF header
+	 */
+	if( length >= 8 &&
+		bytes[0] == 'I' &&
+		bytes[1] == 'I' &&
+		bytes[2] == 42 &&
+		bytes[3] == 0 )
+		ifd_offset = ((guint32) bytes[7] << 24) |
+			((guint32) bytes[6] << 16) |
+			((guint32) bytes[5] << 8) |
+			bytes[4];
+
+	if( length >= 10 &&
+		ifd_offset >= 6 &&
+		bytes[8] == 'C' &&
+		bytes[9] == 'R' )
+		return( "CR2" );
+
+	if( length >= 12 &&
+		ifd_offset >= 8 &&
+		((bytes[8] == 28 &&
+		  bytes[9] == 00 &&
+		  bytes[10] == 254 &&
+		  bytes[11] == 00) ||
+		 (bytes[8] == 31 &&
+		  bytes[9] == 00 &&
+		  bytes[10] == 11 &&
+		  bytes[11] == 00) ||
```

Comment:
FWIW, this was based on:
https://github.com/sindresorhus/file-type/blob/1c75cfbaab544bd0882e498bcbfa54882ea7300e/core.js#L1521

But that unfortunately did not recognize the buffer in the linked issue, which contains `0x00 0x1B 0x00 0xFE` starting from offset 8.

Perhaps there's a better way to recognize these NEF images?

Reply:
https://github.com/libvips/libvips/pull/3460#issuecomment-1615815041